test/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDaoTest.java

```diff
@@ -92,8 +94,24 @@ public void getAllSchedulePlans() {
         schedulePlanDao.createSchedulePlan(abPlan);
         schedulePlanDao.createSchedulePlan(simplePlan);
         
-        List<SchedulePlan> plans = schedulePlanDao.getSchedulePlans(studyIdentifier);
+        List<SchedulePlan> plans = schedulePlanDao.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, studyIdentifier);
         assertEquals("2 plans exist", 2, plans.size());
     }
+    
+    @Test
+    public void filtersSchedulePlans() {
+        for (SchedulePlan plan : TestUtils.getSchedulePlans()) {
+            schedulePlanDao.createSchedulePlan(plan);
+        }
+        
+        // Only one schedule plan matches v9
+        ClientInfo clientInfo = ClientInfo.fromUserAgentCache("app/9");
+        
+        List<SchedulePlan> plans = schedulePlanDao.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, studyIdentifier);
+        assertEquals(3, plans.size());
+        
+        plans = schedulePlanDao.getSchedulePlans(clientInfo, studyIdentifier);
+        assertEquals(1, plans.size());
+    }
```

Comment:
This test depends on the absolute count of schedule plans in DDB. This will break if tests are run in parallel (such as multiple builds in Travis.) It would be better to verify that the list includes the plans you expect (by guid) and excludes the plans you don't.